src/piton/featurizers/featurizers_notes.py

```diff
@@ -0,0 +1,570 @@
+from __future__ import annotations
+
+import multiprocessing
+import os
+import pickle
+import time
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    List,
+    Literal,
+    Optional,
+    Tuple,
+    TypeAlias,
+)
+
+import numpy as np
+import torch
+import transformers
+from nptyping import Int, NDArray
+from torchtyping import TensorType
+from transformers import AutoModel, AutoTokenizer
+
+from .. import Event, Patient
+from ..datasets import PatientDatabase
+from ..labelers.core import Label, LabeledPatients
+from .core import ColumnValue
+
+NotesTokenized: TypeAlias = Dict[
+    str, TensorType["n_notes", "max_note_token_count", int]  # noqa
+]
+NotesEmbedded: TypeAlias = TensorType["n_notes", "embedding_length"]  # noqa
+NotesEmbeddedByToken: TypeAlias = TensorType[
+    "n_notes", "max_note_token_count", "embedding_length"  # noqa
+]
+NotesProcessed = List[Tuple[int, Event]]  # event_idx, event (note)
+PatientLabelNotesTuple = Tuple[
+    int, int, NotesProcessed
+]  # patient_id, label_idx, notes
+
+
+START_TIME = time.time()
+
+
+def print_log(name: str, content: str):
+    print(f"{int(time.time() - START_TIME)} | {name} | {content}")
+
+
+def save_to_pkl(object_to_save, path_to_file: str):
+    """Save object to Pickle file."""
+    os.makedirs(os.path.dirname(path_to_file), exist_ok=True)
+    with open(path_to_file, "wb") as fd:
+        print(fd is None, object_to_save is None)
```

Comment:
Having random print statements like this throughout the codebase tends to be suboptimal